includes/admin/class-amp-story-templates.php

```diff
@@ -0,0 +1,197 @@
+<?php
+/**
+ * AMP Story Templates handler.
+ *
+ * @package AMP
+ * @since 1.?
+ */
+
+/**
+ * Class AMP_Story_Templates
+ */
+class AMP_Story_Templates {
+
+	/**
+	 * Current version of the templates.
+	 *
+	 * @var string
+	 */
+	const STORY_TEMPLATES_VERSION = '0.1';
+
+	/**
+	 * Slug for templates' taxonomy.
+	 */
+	const TEMPLATES_TAXONOMY = 'amp_template';
+
+	/**
+	 * Slug for templates term.
+	 */
+	const TEMPLATES_TERM = 'story-template';
+
+	/**
+	 * Init.
+	 */
+	public function init() {
+		if ( ! post_type_exists( 'amp_story' ) ) {
+			return;
+		}
+
+		add_filter( 'rest_wp_block_query', array( $this, 'filter_rest_wp_block_query' ), 10, 2 );
+
+		$this->register_taxonomy();
+		$this->maybe_import_story_templates();
+	}
+
+	/**
+	 * Import story templates if they haven't been imported previously.
+	 */
+	public function maybe_import_story_templates() {
+		if ( self::STORY_TEMPLATES_VERSION === AMP_Options_Manager::get_option( 'story_templates_version' ) ) {
+			return;
+		}
+		$this->import_story_templates();
+		// @todo This is commented out for testing purposes until the PR gets ready.
+		// AMP_Options_Manager::update_option( 'story_templates_version', self::STORY_TEMPLATES_VERSION );
+	}
+
+	/**
+	 * Import logic for story templates.
+	 */
+	public function import_story_templates() {
+		$templates = self::get_story_templates();
+		foreach ( $templates as $template ) {
+			$existing_template = $this->template_exists( $template['title'], $template['content'] );
+			if ( $existing_template > 0 ) {
+				continue;
+			}
+			$post_id = wp_insert_post(
+				array(
```

Comment:
Note that this array needs to be wrapped in `wp_slash()` unfortunately, as `wp_insert_post()` expects slashed input. If this is not done backslashes will get stripped from the content, and this could be particularly bad for block attributes in JSON.